src/tsdb.c

```diff
@@ -71,6 +73,46 @@ static void seriesEncodeTimestamp(void *buf, timestamp_t timestamp) {
     memcpy(buf, &e, sizeof(e));
 }
 
+void freeLastDeletedSeries() {
+    if(lastDeletedSeries == NULL){
+        return;
+    }
+    CompactionRule *rule = lastDeletedSeries->rules;
+    while (rule != NULL) {
+        CompactionRule *nextRule = rule->nextRule;
+        FreeCompactionRule(rule);
+        rule = nextRule;
+    }
+    if(lastDeletedSeries->srcKey != NULL) {
+        RedisModule_FreeString(NULL, lastDeletedSeries->srcKey);
+    }
+    RedisModule_FreeString(NULL, lastDeletedSeries->keyName);
+    free(lastDeletedSeries);
+    lastDeletedSeries = NULL;
+}
+
+void CleanLastDeletedSeries(RedisModuleCtx *ctx, RedisModuleString *key){
+    if(lastDeletedSeries && RedisModule_StringCompare(lastDeletedSeries->keyName, key) == 0) {
```

Comment:
@gkorland can you make it more explicit?
```suggestion
    if(lastDeletedSeries == NULL && RedisModule_StringCompare(lastDeletedSeries->keyName, key) == 0) {
```

Reply:
@danni-m I guess you meant ```lastDeletedSeries != NULL```